packages/images-plugin/lib/finder.js

```diff
@@ -136,16 +136,23 @@ export default class ImageElementFinder {
     }
 
     const results = [];
-    const condition = async () => {
+    let didFixTemplateImageScale = false;
+    const performLookup = async () => {
       try {
+
         const {screenshot, scale} = await this.getScreenshotForImageFind(driver, screenSize);
 
-        template = await this.fixImageTemplateScale(template, {
-          defaultImageTemplateScale,
-          ignoreDefaultImageTemplateScale,
-          fixImageTemplateScale,
-          ...scale,
-        });
+        if (!didFixTemplateImageScale) {
+          template = await this.fixImageTemplateScale(template, {
+            defaultImageTemplateScale,
+            ignoreDefaultImageTemplateScale,
+            fixImageTemplateScale,
+            ...(scale || {}),
+          });
+          // We do not want `template` to be mutated multiple times when the
```

Comment:
oh, i see. the lambda did multiple times..

Reply:
👍 good catch